### PR TITLE
Attempt to fix arrow 10.0.0 compilation on macOS

### PR DESCRIPTION
**Pull request by jmolnar-comparative**
Gets rid of `cflags!` and `cflags_cc!` (what did these even do?) and moves XCode extra flags to `OTHER_CPLUSPLUSFLAGS`

### CODE REVIEW

File: binding.gyp [hunk 1]
```diff
@@ -9,10 +9,11 @@
...
       ],
       "cflags!": [ "-fno-exceptions", "-fno-rtti" ],
-      "cflags_cc!": [ "-fno-exceptions", "-fno-rtti" ],
-      "cflags_cc": [ "-std=c++17", "-fexceptions", "-frtti" ],
+      "cflags_cc!": [ "-std=gnu++17", "-fno-exceptions", "-fno-rtti" ],
+      "cflags_cc": [
+        "<!@(pkg-config --cflags parquet arrow)", "-Wall", "-g", "-std=c++17", "-fexceptions", "-frtti" ],
       "cflags": [
-          "<!@(pkg-config --cflags parquet arrow) -Wall -g",
+        "<!@(pkg-config --cflags parquet arrow) -Wall -g",
```
Comment (romgrk) [inline binding.gyp:21]: Keep the indentation consistent

File: binding.gyp [hunk 2]
```diff
@@ -9,10 +9,11 @@
         "<!@(node -p \"require('node-addon-api').include\")"
       ],
       "cflags!": [ "-fno-exceptions", "-fno-rtti" ],
-      "cflags_cc!": [ "-fno-exceptions", "-fno-rtti" ],
-      "cflags_cc": [ "-std=c++17", "-fexceptions", "-frtti" ],
+      "cflags_cc!": [ "-std=gnu++17", "-fno-exceptions", "-fno-rtti" ],
+      "cflags_cc": [
+        "<!@(pkg-config --cflags parquet arrow)", "-Wall", "-g", "-std=c++17", "-fexceptions", "-frtti" ],
```
Comment (romgrk) [inline binding.gyp]: Is it absolutely required to add everything before `std=c++17` here?

**Review by romgrk — APPROVED** [review]

File: binding.gyp (hunk 1 above)
Comment (jmolnar-comparative) [reply binding.gyp:21]: That was to make the indentation consistent between the two actually. I'll go ahead and make it **all** consistent though

File: binding.gyp (hunk 2 above)
Comment (jmolnar-comparative) [reply binding.gyp]: double-checked, and it must have been made obsolete by something else I tried. [d365710](https://github.com/Comparative-Inc/comparative-parquet/pull/8/commits/d36571040677307b263a746350bae2eee2fc236b)

File: binding.gyp (hunk 1 above)
Comment (jmolnar-comparative) [reply binding.gyp:21]: https://github.com/Comparative-Inc/comparative-parquet/pull/8/commits/5458a4b66c607f5095ee8ee0bf22893944430528

**Pull request merged**